src/testsets.jl

```diff
@@ -28,56 +24,81 @@ function finish(ts::ConcurrentTestSet)
     if Test.get_testset_depth() > 0
         # Attach this test set to the parent test set
         parent_ts = Test.get_testset()
-        record(parent_ts, ts.dts)
+        record(parent_ts, ts)
         return ts
     end
     finish(ts.dts)
-    return ts.dts
+    return ts
 end
 
 function add_test_ref(testset::ConcurrentTestSet, test_ref)
     return push!(testset.tests, test_ref)
 end
 
-# Handle attempted use outside of a ConcurrentTestSEt
+# Handle attempted use outside of a ConcurrentTestSet
 function add_test_ref(testset::AbstractTestSet, test_ref)
     return fetch(test_ref)
 end
 
-# Wrap a DefaultTestSet. Results are recorded, but not printed if nested=true.
+# Wrap a DefaultTestSet with some behavior specific to @test_rel.
+# 
+# Results are recorded, but not printed if nested=true.
 # This is helpful when used in a ConcurrentTestSet where the parent
-# linkage is lost due to the concurrency
+# linkage is lost due to the concurrency.
+#
+# Additionally the whole test set can be checked for broken-ness, this is
+# needed as a test rel desugars into multiple checks and we don't know which
+# is expected not to work.
 mutable struct TestRelTestSet <: AbstractTestSet
     dts::Test.DefaultTestSet
     nested::Bool
+    broken::Bool
```

Comment:
While you're rearranging, maybe rename this to `broken_expected`? The TestRelTestSet is marked as broken, and the flag is broken_expected.